app/models/nic/bmc.rb

```diff
@@ -39,6 +37,24 @@ def ensure_physical
     def enc_attributes
       @enc_attributes ||= (super + %w(username password provider))
     end
+
+    private
+
+    def get_proxy
```

Comment:
Nitpick, I'd call this `bmc_proxy` to be specific and for consistency with smart proxy associations on other models.